janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/StaticArrayEntryList.java

```diff
@@ -438,15 +440,14 @@ public Boolean get(byte[] array, int offset, int limit) {
 
     private static byte[] ensureSpace(byte[] data, int offset, int length) {
         if (offset+length<=data.length) return data;
```

Comment:
All other cases are handled by this PR except when offset+length overflows. I think it makes sense to throw an exception because the input requires more memory than we could offer using a single byte array.

Reply:
Thank you. Agree. Makes sense.